tritium-caffeine/src/main/java/com/palantir/tritium/metrics/caffeine/CaffeineCacheStats.java

```diff
@@ -37,19 +36,23 @@ private CaffeineCacheStats() {
      * @param cache cache to instrument
      * @param name cache name
      */
-    public static <C extends Cache<?, ?>> void registerCache(MetricRegistry registry, Cache<?, ?> cache, String name) {
+    public static void registerCache(TaggedMetricRegistry registry, Cache<?, ?> cache, String name) {
```

Comment:
or maybe we'd be better off doing the package and artifact rename?

Reply:
I can leave a deprecated version of the old signature that ignores the metric registry and just uses the default tagged registry. Feels a little dirty, but doing that right now in Tritium.instrument, and could do same in Instrumentation.

Reply:
seems reasonable but is a pretty major breaking behavior change. think we'll be alright since we'll log everything from the TaggedMetricRegistry, but should call out in release notes

Reply:
do callers call this directly vs. some Tritium master static class? might be worth leaving the old thing and even the old behavior in place vs. taking the break all at once. ie do the deprecation but actually also leave the old functionality

Reply:
callers do call this if they're using Caffeine caches as opposed to Guava ones instrumented via `MetricRegistries.registerCache`

Reply:
a _very_ gross way to multiplex this would be to have the core impl take a 'register' function and then to have a dropwizard-metrics binding and a tagged-metrics binding that sets function correctly

Reply:
maybe it's not worth it though